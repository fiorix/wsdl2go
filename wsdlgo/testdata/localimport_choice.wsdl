<!-- source: https://www.w3.org/2001/03/14-annotated-WSDL-examples.html -->

<?xml version="1.0"?>

<!-- root element wsdl:definitions defines set of related services -->
<wsdl:definitions name="StockQuote"
             targetNamespace="http://example.com/stockquote.wsdl"
             xmlns:tns="http://example.com/stockquote.wsdl"
             xmlns:xsd1="http://example.com/stockquote.xsd"
             xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/"
             xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/">

    <!-- wsdl:types encapsulates schema definitions of communication types; here using xsd -->
    <wsdl:types>

        <xsd:schema>
            <xsd:import namespace="http://example.com/stockquote.xsd"
                        xmlns:xsd="http://www.w3.org/2000/10/XMLSchema"
                        schemaLocation="testdata/localimport_choice.xsd" />
        </xsd:schema>
    </wsdl:types>

    <!-- request GetLastTradePriceInput is of type Trade-PriceRequest -->
    <wsdl:message name="GetLastTradePriceInput">
        <wsdl:part name="body" element="xsd1:Trade-PriceRequest"/>
    </wsdl:message>

    <!-- request GetLastTradePriceOutput is of type TradePrice -->
    <wsdl:message name="GetLastTradePriceOutput">
        <wsdl:part name="body" element="xsd1:TradePrice"/>
    </wsdl:message>

    <!-- wsdl:portType describes messages in an operation -->
    <wsdl:portType name="StockQuotePortType">

    <!-- the value of wsdl:operation eludes me -->
        <wsdl:operation name="GetLastTradePrice">
           <wsdl:input message="tns:GetLastTradePriceInput"/>
           <wsdl:output message="tns:GetLastTradePriceOutput"/>
        </wsdl:operation>
    </wsdl:portType>

    <!-- wsdl:binding states a serialization protocol for this service -->
    <wsdl:binding name="StockQuoteSoapBinding"
                  type="tns:StockQuotePortType">

        <!-- leverage off soap:binding document style @@@(no wsdl:foo pointing at the soap binding) -->
        <soap:binding style="document" transport="http://schemas.xmlsoap.org/soap/http"/>

        <!-- semi-opaque container of network transport details classed by soap:binding above @@@ -->
        <wsdl:operation name="GetLastTradePrice">

           <!-- again bind to SOAP? @@@ -->
           <soap:operation soapAction="http://example.com/GetLastTradePrice"/>
           <!-- furthur specify that the messages in the wsdl:operation "" use SOAP? @@@ -->
           <wsdl:input>
               <soap:body use="literal"/>
           </wsdl:input>
           <wsdl:output>
               <soap:body use="literal"/>
           </wsdl:output>
        </wsdl:operation>
    </wsdl:binding>

    <!-- wsdl:service names a new service "StockQuoteService" -->
    <wsdl:service name="StockQuoteService">
        <wsdl:documentation>My first service</wsdl:documentation>

        <!-- connect it to the binding "StockQuoteBinding" above -->
        <wsdl:port name="StockQuotePort"
                   binding="tns:StockQuoteBinding">

           <!-- give the binding an network address -->
           <soap:address location="http://example.com/stockquote"/>
        </wsdl:port>
    </wsdl:service>

</wsdl:definitions>
