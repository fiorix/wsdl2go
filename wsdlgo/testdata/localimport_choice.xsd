<?xml version="1.0" encoding="UTF-8"?>	
<xsd:schema targetNamespace="http://example.com/stockquote.xsd"
              xmlns:xsd="http://www.w3.org/2000/10/XMLSchema">


    <!-- xsd definition: TradePriceRequest [... tickerSymbol string ...] -->
    <xsd:element name="TradePriceRequest">
        <xsd:complexType>
            <xsd:all>
                <xsd:element name="tickerSymbol" type="string"/>
            </xsd:all>
        </xsd:complexType>
    </xsd:element>


    <!-- xsd definition: TradePrice choice ( price float, or discountprice) -->
    <xsd:element name="TradePrice">
        <xsd:complexType>
            <xsd:choice>
                <xsd:element name="price" type="float"/>
		<xsd:element name="discountprice" type="float"/>
            </xsd:choice>
        </xsd:complexType>
    </xsd:element>
</xsd:schema>
