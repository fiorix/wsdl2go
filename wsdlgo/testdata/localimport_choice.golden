package stockquotesoapbinding

import (
	"github.com/fiorix/wsdl2go/soap"
)

// Namespace was auto-generated from WSDL.
var Namespace = "http://example.com/stockquote.wsdl"

// NewStockQuotePortType creates an initializes a StockQuotePortType.
func NewStockQuotePortType(cli *soap.Client) StockQuotePortType {
	return &stockQuotePortType{cli}
}

// StockQuotePortType was auto-generated from WSDL
// and defines interface for the remote service. Useful for testing.
type StockQuotePortType interface {
	// GetLastTradePrice was auto-generated from WSDL.
	GetLastTradePrice(body *TradePriceRequest) (*TradePrice, error)
}

// TradePrice was auto-generated from WSDL.
type TradePrice struct {
	Price         *float64 `xml:"price,omitempty" json:"price,omitempty" yaml:"price,omitempty"`
	Discountprice *float64 `xml:"discountprice,omitempty" json:"discountprice,omitempty" yaml:"discountprice,omitempty"`
}

// TradePriceRequest was auto-generated from WSDL.
type TradePriceRequest struct {
	TickerSymbol *string `xml:"tickerSymbol,omitempty" json:"tickerSymbol,omitempty" yaml:"tickerSymbol,omitempty"`
}

// Operation wrapper for GetLastTradePrice.
// OperationGetLastTradePriceInput was auto-generated from WSDL.
type OperationGetLastTradePriceInput struct {
	Body *TradePriceRequest `xml:"body,omitempty" json:"body,omitempty" yaml:"body,omitempty"`
}

// Operation wrapper for GetLastTradePrice.
// OperationGetLastTradePriceOutput was auto-generated from WSDL.
type OperationGetLastTradePriceOutput struct {
	Body *TradePrice `xml:"body,omitempty" json:"body,omitempty" yaml:"body,omitempty"`
}

// stockQuotePortType implements the StockQuotePortType interface.
type stockQuotePortType struct {
	cli *soap.Client
}

// GetLastTradePrice was auto-generated from WSDL.
func (p *stockQuotePortType) GetLastTradePrice(body *TradePriceRequest) (*TradePrice, error) {
	α := struct {
		OperationGetLastTradePriceInput `xml:"tns:GetLastTradePrice"`
	}{
		OperationGetLastTradePriceInput{
			body,
		},
	}

	γ := struct {
		OperationGetLastTradePriceOutput `xml:"GetLastTradePriceResponse"`
	}{}
	if err := p.cli.RoundTripWithAction("http://example.com/GetLastTradePrice", α, &γ); err != nil {
		return nil, err
	}
	return γ.Body, nil
}
